netlogo-gui/src/main/lab/gui/ProgressDialog.scala

```diff
@@ -139,6 +136,26 @@ private [gui] class ProgressDialog(dialog: java.awt.Dialog, supervisor: Supervis
       workspace.jobManager.finishSecondaryJobs(null)
     }
   }
+
+  def updateView(check: Boolean): Unit = {
+    displaySwitch.setSelected(check)
+    workspace.displaySwitchOn(check)
+  }
+
+  def setUpdateView(status: Boolean): Unit = {
+    updateView(status)
+  }
+
+  def plotsAndMonitorsSwitch(check: Boolean): Unit = {
+    plotsAndMonitorsSwitch.setSelected(check)
+    workspace.setPeriodicUpdatesEnabled(check)
```

Comment:
You can add in a conditional call to `workspace.jobManager.finishSecondaryJobs(null)` at the end of this method.

Reply:
There's no need of making that call , I've tested.

Reply:
Just because you don't see a difference doesn't mean that the call isn't doing anything. That call shuts down jobs which run on the background thread. For models with expensive-to-compute monitors, `finishSecondaryJobs` will have a major impact.

Reply:
Okay, will make that change too.